tools/dockerfreeze

```diff
@@ -1,15 +1,16 @@
-#!/bin/sh
+#!/bin/false
+# This is BASH completion file expected to be sourced, do not invoke it
+
 _dockerfreeze() {
-    local cur prev opts cmds
-    COMREPLY=()
+    local cur opts
+    COMPREPLY=()
     cur="${COMP_WORDS[COMP_CWORD]}"
-    prev="${COMP_WORDS[COMP_CWORD-1]}"
     opts="-g -h -v -o --version --help --gitpod --output"
     if [[ ${cur} == -* ]] ; then
-        COMPREPLY=( $(compgen -W "${opts}" -- ${cur}) )
+        mapfile -t COMPREPLY < <(compgen -W "${opts}" -- "${cur}")
         return 0
     elif [[ ${cur} == r* ]]; then
-        COMPREPLY=( $(compgen -W "${opts}" ${cur}) )
+        mapfile -t COMPREPLY < <(compgen -W "${opts}" "${cur}")
         return 0
     fi
```

Comment:
See http://dpaste.com/1R88CM5.txt

It's more efficient to use case statements over double square brackets for this runtime

This is not scientific enough though might require more tests to confirm the performance last time i was investigating this it was by 0.0052 sec faster

Reply:
EDIT: i am reading it wrong it's slower by up to 0.0012 sec in this case.

Reply:
Ok